src/Rocketlet.ts

```diff
@@ -144,7 +144,7 @@ export abstract class Rocketlet implements IRocketlet {
      * Method will be called during initialization. It allows for adding custom configuration options and defaults
      * @param configuration
      */
-    protected extendConfiguration(configuration: IConfigurationExtend): void {
+    protected extendConfiguration(configuration: IConfigurationExtend, environmentRead: IEnvironmentRead): void {
```

Comment:
When creating settings, they might depend on other settings or env-variables => One might want to read them for the default values.